docs/design/proposals/encrypted-pvc.md

```diff
@@ -127,3 +127,20 @@ data:
 metadata:
   name: ceph-csi-encryption-kms-config
 ```
+
+### Implementation Details
+
+The main components that are used to support encrypted volumes:
```

Comment:
@nixpanic eventhough we are bit away from supporting cephfs encryption, just wondering how these methods and implementation adjust with when its available, not a concern for now, but good to consider as well.

Reply:
we don't know yet how a CephFS encryption interface would look like, but I am pretty confident that this split of encryption and DEK-storage does not complicate things for that

Reply:
CephFS encryption will be done with `fscrypt` library (https://www.kernel.org/doc/html/v4.19/filesystems/fscrypt.html)  support,  this would be different from the way we make use of rbd image metadata .etc. #1563 ..Considering CephFS encryption is wip, lets move ahead with current version here.